react/player/src/manager/managed-player.tsx

```diff
@@ -284,6 +286,12 @@ export const ManagedPlayer = (props: ManagedPlayerProps) => {
     };
   }, [props.manager, state?.context.reactPlayer.player, state?.value]);
 
+  React.useEffect(() => {
+    if (managedState.state) {
+      managedState.state.context.manager = props.manager;
+    }
+  }, [props.manager]);
```

Comment:
IMO this feels like we should be terminating the old managed-player before picking up with the new one?

Reply:
Yeah this was just me hacking things together to test if this was the source of the issue. In theory a new Flow Manager should only be passed when managed player is in a `completed` state before the `pending` state builds the `next` function called in the `pending` state. If the parent app wants to override the initial Flow Manager it started player with, it should only be able to do so between flows.